criu/parasite-syscall.c

```diff
@@ -262,14 +260,10 @@ int parasite_dump_sigacts_seized(struct parasite_ctl *ctl, struct pstree_item *i
 		ASSIGN_TYPED(sa->sigaction, encode_pointer(args->sas[i].rt_sa_handler));
 		ASSIGN_TYPED(sa->flags, args->sas[i].rt_sa_flags);
 		ASSIGN_TYPED(sa->restorer, encode_pointer(args->sas[i].rt_sa_restorer));
-#ifdef CONFIG_MIPS
-		sa->has_mask_extended = 1;
-		BUILD_BUG_ON(sizeof(sa->mask) * 2 != sizeof(args->sas[0].rt_sa_mask.sig));
-		memcpy(&sa->mask, &(args->sas[i].rt_sa_mask.sig[0]), sizeof(sa->mask));
-		memcpy(&sa->mask_extended, &(args->sas[i].rt_sa_mask.sig[1]), sizeof(sa->mask));
-#else
-		BUILD_BUG_ON(sizeof(sa->mask) != sizeof(args->sas[0].rt_sa_mask.sig));
 		memcpy(&sa->mask, args->sas[i].rt_sa_mask.sig, sizeof(sa->mask));
+#ifdef ARCH_HAS_SIG_MASK_EXTENDED
```

Comment:
I strongly agree with you about it looks safer to keep `BUILD_BUG_ON`, but what I’m thinking about is whether it's worth it.

I don't think this will come into effect when maintainer of different architectures complete the work of porting. Of course, my knowledge of criu is still shallow. I'd be happy to ask your further views.